src/test/java/uk/ac/cam/cl/dtg/segue/api/AdminFacadeTest.java

```diff
@@ -0,0 +1,423 @@
+import uk.ac.cam.cl.dtg.segue.dao.content.ContentManagerException;
+import uk.ac.cam.cl.dtg.segue.dao.content.GitContentManager;
+import uk.ac.cam.cl.dtg.segue.scheduler.SegueJobService;
+import uk.ac.cam.cl.dtg.util.PropertiesLoader;
+
+class AdminFacadeTest {
+  private PropertiesLoader properties;
+  private UserAccountManager userManager;
+  private EmailManager emailManager;
+  private GitContentManager contentManager;
+  private String contentIndex;
+  private ILogManager logManager;
+  private StatisticsManager statsManager;
+  private AbstractUserPreferenceManager userPreferenceManager;
+  private EventBookingManager eventBookingManager;
+  private SegueJobService segueJobService;
+  private IExternalAccountManager externalAccountManager;
+  private IMisuseMonitor misuseMonitor;
+  private AdminFacade adminFacade;
+
+  @BeforeEach
+  final void beforeEach() {
+    properties = createMock(PropertiesLoader.class);
+    userManager = createMock(UserAccountManager.class);
+    contentManager = createMock(GitContentManager.class);
+    contentIndex = "";
+    logManager = createMock(ILogManager.class);
+    statsManager = createMock(StatisticsManager.class);
+    userPreferenceManager = createMock(AbstractUserPreferenceManager.class);
+    eventBookingManager = createMock(EventBookingManager.class);
+    segueJobService = createMock(SegueJobService.class);
+    externalAccountManager = createMock(IExternalAccountManager.class);
+    misuseMonitor = createMock(IMisuseMonitor.class);
+    emailManager = createMock(EmailManager.class);
+    adminFacade = new AdminFacade(properties, userManager, contentManager, contentIndex, logManager, statsManager,
+        userPreferenceManager, eventBookingManager, segueJobService, externalAccountManager, misuseMonitor, emailManager);
+  }
+
+  @Nested
+  class ModifyUsersTeacherPendingStatus {
+    @Nested
+    class FailureResponses {
+      @Test
+      void modifyUsersTeacherPendingStatus_anonymousUser_returnsUnauthorised() throws NoUserLoggedInException {
+        HttpServletRequest mockRequest = replayMockServletRequest();
+        List<Long> targetUsers = List.of(2L);
+
+        expect(userManager.getCurrentRegisteredUser(mockRequest)).andThrow(new NoUserLoggedInException());
+
+        replay(userManager);
+
+        try (Response response = adminFacade.modifyUsersTeacherPendingStatus(mockRequest, false, targetUsers)) {
+          assertEquals(Response.Status.UNAUTHORIZED.getStatusCode(), response.getStatus());
+          assertEquals("You must be logged in to access this resource.",
+              response.readEntity(SegueErrorResponse.class).getErrorMessage());
+        }
+
+        verify(userManager);
+      }
+
+      @Test
+      void modifyUsersTeacherPendingStatus_nonAdminUser_returnsForbidden() throws NoUserLoggedInException {
+        HttpServletRequest mockRequest = replayMockServletRequest();
+        RegisteredUserDTO currentUser = new RegisteredUserDTO();
```

Comment:
It sounds like in a lot of these tests you could have simply declared the `currentUser` as a mock, since you don't really need to interact with it at all, and you don't care about its contents, nor any of its methods need to return anything in particular. It's not a big deal in this case, since it's a DTO, so there is no business logic inside anyway, but still something to think about. Making it a mock would also signal more clearly that you don't really care about this user, and it's only here because it's needed as a parameter of other functions.